presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java

```diff
@@ -139,12 +142,15 @@ private void killContainersReaperContainer()
             }
         }
 
-        private void sleepUntilInterrupted()
+        private void sleepUntilInterruptedOrContainersStopped(Collection<DockerContainer> containers)
         {
             try {
-                //noinspection InfiniteLoopStatement
                 while (true) {
-                    Thread.sleep(Long.MAX_VALUE);
+                    Thread.sleep(10_000);
+                    if (containers.stream().noneMatch(ContainerState::isRunning)) {
```

Comment:
can we use `failsafe` for this?

Reply:
I don't think we should. failsafe is supposed to be used for a one-off action that may need to be retried on failure.
Here we have a condition that, when satisfied, is actually an error condition.

Reply:
The pattern with throwing an exception when the condition passes doesn't fit into failsafe neatly so I switched it back. 

I also moved the `isRunning` condition back here. I think that passing in a predicate was bad for readability, since it separated the condition from its usage.